waqi_client_async/__init__.py

```diff
@@ -29,30 +33,51 @@ class OverQuota(APIError):
     pass
 
 
+class UnknownCity(APIError):
+    """Raised when the provided city could not be found."""
+
+    pass
+
+
+class UnknownID(APIError):
+    """Raised when the provided ID could not be found."""
+
+    pass
+
+
 class UnknownStation(APIError):
     """Raised when the provided station could not be found."""
 
     pass
 
 
 def assert_valid(result: dict) -> None:
-    if result.get("status") != "error":
-        return
+    LOGGER.debug("JSON Data: %s", result)
+    if result.get("status") == "ok":
+        if data := result.get("data") is not None:
+            if data.get("msg") == "Unknown ID":
+                raise UnknownID()
 
-    data = result.get("data")
-    if data == "Invalid key":
+    return
```

Comment:
This means the code below does not get executed.

Reply:
Fixed, needs one more ident.